src/test/java/com/google/sps/SurveyServletAndResponseTest.java

```diff
@@ -0,0 +1,413 @@
+package com.google.sps.servlets;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.IncompleteKey;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.PathElement;
+import com.google.cloud.datastore.testing.LocalDatastoreHelper;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/** 
+ * Tests for the SurveyServlet and SurveyResponse classes.
+ *
+ * Make sure to set the project you're working on in your shell with:
+ * gcloud config set project [PROJECT_NAME]
+ *
+ * Skeleton code for Datastore emulator from:
+ * https://stackoverflow.com/questions/40348653/google-datastore-emulator-using-java-not-using-gae
+ */
+public class SurveyServletAndResponseTest {
```

Comment:
Are all of these tests for the same class?

Reply:
They're not - I put them in the same class because they use the same data and I didn't want to copy it twice. But after some googling I found I can use inheritance so two test classes can share a superclass with the data. Implemented in cf7531e068ee1ac9ef6a4b4c7854cc482ad8ed31.